packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -20,6 +21,20 @@ export interface HttpGatewayListenerOptions {
    * @default - no healthcheck
    */
   readonly healthCheck?: HealthCheck;
+
+  /**
+   * The TLS mode.
+   *
+   * @default - none
+   */
+  readonly tlsMode?: TlsMode;
```

Comment:
Why can `TlsMode` be set both in `TlsCertificate`, and here, on the listener level...?

Reply:
In the App Mesh listener spec a TLS struct is used for the mode and certificate shapes. But to better follow the [design guidelines](https://github.com/aws/aws-cdk/blob/master/DESIGN_GUIDELINES.md#flat) I pulled these fields out to flatten the CDK equivalent code.

This is how the shape appears in our spec:

```yaml
Listeners: [
  TLS: {
    Mode: "STRICT",
    Certificate: {
      ACM: {
        CertificateArn: "arn:..."
      },
    },
  },
],
```

And here's how that^ config would look flattened in CDK.

```yaml
listeners: [appmesh.VirtualNodeListener.grpc({
  tlsCertificate: appmesh.TlsCertificate.acm({
    tlsMode: TlsMode.STRICT,
    acmCertificate: cert,
  }),
},
```

Reply:
Right. Except with the way the code is written now, you can also do this:

```ts
listeners: [appmesh.VirtualNodeListener.grpc({
  tlsMode: TlsMode.DISABLED,
  tlsCertificate: appmesh.TlsCertificate.acm({
    tlsMode: TlsMode.STRICT,
    acmCertificate: cert,
  }),
},
```

Why can you set `tlsMode` in 2 places when defining the listener? Do they have different semantics somehow?

Reply:
ah I see, yeah it shouldn't be able to be set at the listener level. Removed.